graylog2-server/src/main/java/org/graylog2/configuration/converters/JavaDurationConverter.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog2.configuration.converters;
+
+import com.github.joschi.jadconfig.Converter;
+
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.concurrent.TimeUnit;
+
+public class JavaDurationConverter implements Converter<Duration> {
+    @Override
+    public Duration convertFrom(String value) {
+        final com.github.joschi.jadconfig.util.Duration jadDuration = com.github.joschi.jadconfig.util.Duration.parse(value);
+
+        final ChronoUnit chronoUnit = toChronoUnit(jadDuration.getUnit());
+        return Duration.of(jadDuration.getQuantity(), chronoUnit);
```

Comment:
What's wrong with `toChronoUnit()`? This way it's a generic converter that doesn't loose nanosecond precision and also supports very large durations.

Reply:
@thll I thought we can make this a bit simpler by avoiding the large switch statement. But I didn't consider the precision, of course... :sweat_smile: Since this is a generic converter, we should leave it as it is. :+1: